dockstore-webservice/src/main/java/io/dockstore/webservice/core/database/EntryLite.java

```diff
@@ -0,0 +1,108 @@
+package io.dockstore.webservice.core.database;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import io.dockstore.common.EntryType;
+import io.dockstore.common.SourceControl;
+import io.dockstore.webservice.core.BioWorkflow;
+import io.dockstore.webservice.core.Service;
+import io.dockstore.webservice.core.Tool;
+
+/**
+ * This class describes lightweight entry objects that are used for making type-safe named queries
+ * @author ldcabansay
+ * @since 1.9.0
+ */
+public abstract class EntryLite {
+    private final Date lastUpdated;
+
+    EntryLite(Date entryUpdated, Date versionUpdated) {
+        //choose the greater update time between overall entry and most recently updated version
+        this.lastUpdated = versionUpdated == null ? entryUpdated : (entryUpdated.getTime() > versionUpdated.getTime() ? entryUpdated : versionUpdated);
+    }
+
+    public Date getLastUpdated() {
+        return lastUpdated;
+    }
+
+    public abstract String getEntryPath();
+
+    public abstract EntryType getEntryType();
+
+    public String makePrettyPath(String path) {
```

Comment:
It looks like the only place you call this, you're passing `getEntryPath()` as a parameter. Since `getEntryPath()` is already a class method, I would get rid of the parameter and just have `makePrettyPath` use `getEntryPath`